kWidget/onPagePlugins/omnitureOnPage/resources/omnitureOnPage.js

```diff
@@ -463,60 +478,80 @@ kWidget.addReadyCallback( function( playerId ){
 			}
 		},
 		runMediaCommandWithArgs: function( args ){
-			var cmd = args[0];
-	 		var argSet = args.slice( 1 );
-	 		
-	 		var s = window[ this.getSCodeName() ];
-	 		try {
-	 			// When using argSet.join we turn all arguments to string, we need to send them with the same type 
-	 			//eval( this.getSCodeName() + '.Media.' + cmd + '("' + argSet.join('","') + '");');
-	 			// not working :(
-	 			//s.Media[cmd].apply( this, args );
 
-				if(this.getConfig("overridePlayerName") != undefined ){
-					s.Media.playerName = String(this.getConfig("overridePlayerName"));
-				}
-			    if ( !s.inAd && cmd !== 'openAd') {
-				    // re-evaluate mediaName even it it was already pushed to the stack
-				    argSet[0] = this.getMediaName();
-			    }
-		 		switch( cmd ) {
-		 			case 'open':
-						s.Media.open(argSet[0], argSet[1], argSet[2]);
-		 			break;
-		 			case 'play':
-		 				s.Media.play(argSet[0], argSet[1]);
-		 			break;
-		 			case 'stop':
-		 				s.Media.stop(argSet[0], argSet[1]);
-		 			break;
-		 			case 'close':
-		 				s.Media.close(argSet[0]);
-					    s.inAd = false;
-		 			break;
-					case 'openAd':
-						s.inAd = true;
-		 				s.Media.openAd(argSet[0], argSet[1], argSet[2],argSet[3], argSet[4], argSet[5]);
-		 			break;
-					case 'complete':
-						s.Media.complete(argSet[0], argSet[1]);
+
+			var eventsTimeout = 0; //eventsTimeout is a new configuration, delay time in seconds
+			if (this.getConfig("eventsTimeout")){
+					eventsTimeout = this.getConfig("eventsTimeout")*1000;
+			}
+			_this = this;
+			var _args = args;
+
+
+			setTimeout(function(){
+
+
+				var s = window[ _this.getSCodeName() ];
+				var cmd = _args[0];
+				var argSet = _args.slice( 1 );
+				try {
+					// When using argSet.join we turn all arguments to string, we need to send them with the same type
+					//eval( this.getSCodeName() + '.Media.' + cmd + '("' + argSet.join('","') + '");');
+					// not working :(
+					//s.Media[cmd].apply( this, args );
+
+					if(_this.getConfig("overridePlayerName") != undefined ){
+						s.Media.playerName = String(_this.getConfig("overridePlayerName"));
+					}
+					if ( !s.inAd && cmd !== 'openAd') {
+						// re-evaluate mediaName even it it was already pushed to the stack
+						argSet[0] = _this.getMediaName();
```

Comment:
previousName is set only once on Open so only a media-open can change it. Since I don't see ads and timeout scenario now I will just add a configuration that I will turn on only when needed. What do you think of this?